Gems/Atom/RHI/DX12/Code/Source/RHI/AsyncUploadQueue.cpp

```diff
@@ -243,232 +241,249 @@ namespace AZ
-            {
-                AZ_PROFILE_SCOPE(RHI, "Upload Image");
-                ID3D12CommandQueue* dx12CommandQueue = static_cast<ID3D12CommandQueue*>(commandQueue);
-                FramePacket* framePacket = BeginFramePacket();
+                uint32_t startMip = residentMip - 1;
+                uint32_t endMip = residentMip - static_cast<uint32_t>(request.m_mipSlices.size());
 
-                uint32_t arraySize = request.m_image->GetDescriptor().m_arraySize;
-                uint16_t imageMipLevels = request.m_image->GetDescriptor().m_mipLevels;
+                Memory* imageMemory = static_cast<Image&>(*request.m_image).GetMemoryView().GetMemory();
 
-                // Variables for split subresource slice. 
-                // If a subresource slice pitch is large than one staging size, we may split the slice by rows.
-                // And using the CopyTextureRegion to only copy a section of the subresource 
-                bool needSplitSlice = false;
-                uint32_t rowsPerSplit = 0;
+                const RHI::StreamingImageExpandRequest& cachedRequest = m_imageExpandRequests.back();
 
-                for (uint32_t curMip = endMip; curMip <= startMip; curMip++)
+                m_copyQueue->QueueCommand([=](void* commandQueue)
                 {
-                    size_t sliceIndex = curMip - endMip;
-                    const RHI::ImageSubresourceLayout& subresourceLayout = request.m_mipSlices[sliceIndex].m_subresourceLayout;
-                    uint32_t arraySlice = 0;
-                    const uint32_t subresourceSlicePitch = subresourceLayout.m_bytesPerImage;
-
-                    // Staging sizes
-                    uint32_t stagingRowPitch = RHI::AlignUp(subresourceLayout.m_bytesPerRow, DX12_TEXTURE_DATA_PITCH_ALIGNMENT);
-                    uint32_t stagingSlicePitch = RHI::AlignUp(subresourceLayout.m_rowCount*stagingRowPitch, D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT);
-                    const uint32_t compressedTexelBlockSizeHeight = subresourceLayout.m_blockElementHeight;
-
-                    // ImageHeight must be bigger than or equal to the Image's row count. Images with a RowCount that is less than the ImageHeight indicates a block compression.
-                    // Images with a RowCount which is higher than the ImageHeight indicates a planar image, which is not supported for streaming images.
-                    if (subresourceLayout.m_size.m_height < subresourceLayout.m_rowCount)
-                    {
-                        AZ_Error("StreamingImage", false, "AsyncUploadQueue::QueueUpload expects ImageHeight '%d' to be bigger than or equal to the image's RowCount '%d'.", subresourceLayout.m_size.m_height, subresourceLayout.m_rowCount);
-                        return 0;
-                    }
+                    AZ_TracePrintf("async upload", "Expand image %s from mip %d to %d\n", cachedRequest.m_image->GetName().GetCStr(), endMip, startMip);
```

Comment:
It seems like this should be removed or disabled by default.

Reply:
I removed it